src/easter-egg.ts

```diff
@@ -0,0 +1,284 @@
+   *     current page with it.
+   * @param {ContinueAfterCallbackFxn} [continueAfterCallback] Whether or not the caller should
+   *     continue whatever action it was in the middle of after a decryption and callback. If
+   *     unspecified, the default function unconditionally returns false.
+   */
+  constructor(params: EasterEggParams, callback?: Callback,
+      continueAfterCallback?: ContinueAfterCallbackFxn) {
+    this.hash = params.hash;
+    this.iv = EasterEgg.hexStrToArrayBuffer(params.iv);
+    this.derivedKeySalt = EasterEgg.hexStrToArrayBuffer(params.derivedKeySalt);
+    this.passwordSalt = EasterEgg.hexStrToArrayBuffer(params.passwordSalt);
+    this.continueAfterCallback = continueAfterCallback ? continueAfterCallback : async () => false;
+
+    if (typeof params.ciphertext === 'function') {
+      this.getCiphertext = params.ciphertext;
+    } else {
+      this.getCiphertext = async (): Promise<ArrayBuffer> =>
+        EasterEgg.hexStrToArrayBuffer(params.ciphertext as HexString);
+    }
+
+    if (!callback) {
+      callback = async (plaintext: ArrayBuffer): Promise<boolean> => {
+        document.body.innerHTML = EasterEgg.textDecoder.decode(plaintext);
+        return true;
+      };
+    }
+    this.callback = callback;
+  }
+
+  /**
+   * Determines if |blob| is the correct password. If so, then decrypts the ciphertext and hands
+   * over the plaintext to the callback.
+   *
+   * @param {Blob} blob The attempted password.
+   * @return {Promise<boolean | undefined>} If undefined, this indicates that the blob's hash did
+   *     not match. Otherwise, the boolean return value indicates whether or not the caller should
+   *     continue with its previous action.
+   */
+  async hunt(blob: Blob): Promise<boolean | undefined> {
+    const blobBuffer: ArrayBuffer = await blob.arrayBuffer();
+    const hash = await EasterEgg.hashWithSalt(blobBuffer, this.passwordSalt);
+    if (hash !== this.hash) {
+      return undefined;
+    }
+
+    const decryptionKey: CryptoKey = await EasterEgg.keyFromPassword(blobBuffer, 'decrypt',
+        this.derivedKeySalt);
+    const plaintext: ArrayBuffer = await window.crypto.subtle.decrypt({
+      name: EasterEgg.encryptionAlgo,
+      iv: this.iv,
+    }, decryptionKey, await this.getCiphertext());
+
+    const callbackSuccess = await this.callback(plaintext);
+    return await this.continueAfterCallback(callbackSuccess);
+  }
+
+  /**
+   * Concatenates a password with a salt, and hashes the result.
+   *
+   * @param {ArrayBuffer} password The password to salt and hash.
+   * @param {ArrayBuffer} salt The salt to password and hash.
+   * @return {Promise<HexString>} Promise to the salted hash of the password.
+   */
+  private static async hashWithSalt(password: ArrayBuffer, salt: ArrayBuffer): Promise<HexString> {
+    const saltedPW: ArrayBuffer = (new Uint8Array([...new Uint8Array(password)]
+        .concat([...new Uint8Array(salt)]))).buffer;
```

Comment:
nit: this would probably be worth splitting up in multiple lines for readability.

Reply:
done